workflows/diagnostics/tests/offline/test_helpers.py

```diff
@@ -112,12 +114,28 @@ def test_insert_column_integrated_vars():
     xr.testing.assert_allclose(insert_column_integrated_vars(ds, ["Q1"]), expected)
 
 
+def test_load_grid_info_from_custom_catalog(datadir):
+    # This test is dedicated to validate the response of "load_grid_info" function
+    # on a bogus catalog file. Retrieving the grid information with a non-existing
+    # resolution string generates the "KeyError".
+
+    res = "c12"
+    catalog_path = os.path.join(datadir, "catalog_dummy.yaml")
+
+    with pytest.raises(KeyError):
+        load_grid_info(res, catalog_path)
```

Comment:
This test needs to be updated based on the new signature of `load_grid_info`:
```suggestion
    catalog_path = os.path.join(datadir, "catalog_dummy.yaml")
    catalog = intake.open_catalog(catalog_path)

    with pytest.raises(KeyError):
        load_grid_info(catalog, res)
```